File: WebApplication/src/app/_orchestration/searchEvents/searchEventsRequestDto.ts
import { removeUndefinedKeyValues } from "../../_common/jsonHelpers"
import { RaceSeriesType } from "../../_core/enums/raceSeriesType"

export class SearchEventsRequestDto {
  constructor(
    public searchTerm: string | null,
    public state: string | null = null,
    public area: string | null = null,
    public city: string | null = null,
    public raceSeriesTypes: RaceSeriesType[] = [],
    public isUpcoming: boolean = false,
  ) {}

  public getAsParamsObject = () => {

    const params = {
      searchTerm: this.searchTerm,
      raceSeriesTypes: this.raceSeriesTypes,
      state: this.state,
      area: this.area,
      city: this.city,
      isUpcoming: this.isUpcoming,
    }

    return removeUndefinedKeyValues(params)
  }
}

File: Api/Orchestration/SearchEvents/SearchEventsRequestDto.cs
using Api.DataModels;

namespace Api.Orchestration.SearchEvents;

public class SearchEventsRequestDto
{
    public string? SearchTerm { get; set; }

    public List<RaceSeriesType> RaceSeriesTypes { get; set; }

    public string? State { get; set; }

    public string? Area { get; set; }

    public string? City { get; set; }

    public SearchEventsRequestDto()
    {
        RaceSeriesTypes = new List<RaceSeriesType>();
    }

    public SearchEventsRequestDto(params RaceSeriesType[] raceSeriesTypes)
    {
        RaceSeriesTypes = raceSeriesTypes.ToList();
    }
}
